Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift

```diff
@@ -108,18 +109,25 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
     }
 
     func googleLogin() {
-        let loginModel = GoogleWeb3AuthLoginModel(authorizationComplete: {
+        let authorizationComplete = {
             let view = self.factory.safeCreatingViewController()
             view.onSuccess = {
                 self.stop(success: true)
             }
             self.show(view)
-
-        } , keyGenerationComplete: { [weak self] key, email in
-            self?.storeKeyAndCreateSafe(key: key, email: email, keyType: .web3AuthGoogle)
-        })
-
-        loginModel.loginWithCustomAuth()
+        }
+        let keyGenerationComplete = { (key, email) in
+            self.storeKeyAndCreateSafe(key: key, email: email, keyType: .web3AuthGoogle)
+        }
+        if let loginModel = loginModel {
+            loginModel.authorizationComplete = authorizationComplete
+            loginModel.keyGenerationComplete = keyGenerationComplete
+        } else {
+            loginModel = GoogleWeb3AuthLoginModel(authorizationComplete: authorizationComplete,
```

Comment:
I would refactor GoogleWeb3AuthLoginModel and make the initializer takes no parameters in this case, this will improve the code her

Reply:
Done